shap/explainers/tree.py

```diff
@@ -366,6 +385,150 @@ def recurse(node):
 
         recurse(0)
 
+    def gen_trees(self, model):
```

Comment:
All this stuff should be in the Tree Object and based on the JSON parsing but I can change that after merging since I already have that implemented.